index.js

```diff
@@ -1,130 +1,163 @@
-    .some(line => argsRegex.test(line.trim()));
-}
-
-function getTemplate(needArgs) {
-  const templateFile = needArgs
-    ? templateWithArgsPath
-    : templatePath;
-
-  return readFile(templateFile, 'utf-8');
-}
-
-function createElmFile(template) {
-  const contents = template.replace('{module}', moduleName);
-
-  return writeFile(outputElmFilename, contents, 'utf-8');
-}
+(async () => {
+  program
+    // package.json is one level up because the compiled js file is in lib/
+    // eslint-disable-next-line import/no-unresolved
+    .version(require(path.resolve(__dirname, '../package.json')).version)
+    .usage('[options] <file> [arg1 arg2 ...]')
+    .description('Runs an Elm file and prints the result of `output` to stdout.'
+      + ' `output` can be a constant of type `String` or a function that accepts `List String` and returns `String`. You can supply command line arguments to the Elm file if `output` accepts `List String`.')
+    .option(
+      '--output-name [name]',
+      'constant or function name for printing results',
+      'output',
+    )
+    .option(
+      '--project-dir [path]',
+      'specific directory to search for elm-package.json or to create it (if different from Elm file location)'
+    )
+    .parse(process.argv);
+
+  // run-elm expects one or more arguments, so exit if no arguments given
+  if (program.args.length === 0) {
+    console.log(program.help());
+    process.exit(1);
+  }
 
-async function compileElmFile() {
-  sh.cd(outputDir);
+  // read args and options
+  const rawUserModuleFileName = program.args[0];
+  const outputName = program.outputName;
+  const rawProjectDir = program.projectDir;
+  const argsToOutput = program.args.slice(1);
+
+  // extract key paths
+  const timestamp = +new Date();
+  const mainModule = `RunElmMain${timestamp}`;
+  const templatePath = path.resolve(__dirname, 'RunElmMain.elm.template');
+  const templateWithArgsPath = path.resolve(__dirname, 'RunElmMainWithArgs.elm.template');
+  const userModulePath = path.resolve(rawUserModuleFileName);
+  const userModule = path.basename(userModulePath, '.elm');
+  const projectDir = rawProjectDir
+    ? path.resolve(rawProjectDir)
+    : path.resolve(path.dirname(userModulePath));
+  const mainModuleFilename = path.join(projectDir, `${mainModule}.elm`);
+  const outputCompiledFilename = path.join(projectDir, `run-elm-main-${timestamp}.js`);
+
+  let exitCode = 0;
+  try {
+    // ensure elm is installed
+    if (!sh.which('elm')) {
+      throw new Error('No elm global binary available. Please install with `npm install -g elm`.');
+    }
 
-  const contents = await compileToString([outputElmFilename], { yes: true });
+    // ensure --output-name is specified adequately
+    if (!outputName.match(/^[a-z_]\w*$/)) {
+      throw new Error(`Provided --output-name \`${outputName}\` is not a valid constant or function name in elm.`);
+    }
+    if (['init', 'main', 'program', 'sendOutput'].indexOf(outputName) !== -1) {
```

Comment:
`Array.prototype.includes` should be available in Node 6.

Reply:
True: http://node.green/#ES2016-features-Array-prototype-includes